pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -400,7 +410,7 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 		"--fsid", c.clusterInfo.FSID,
 		"--setuser", "ceph",
 		"--setgroup", "ceph",
-		fmt.Sprintf("--crush-location=%s", osd.Location),
+		fmt.Sprintf("--crush-location=%q", osd.Location),
```

Comment:
Why are quotes needed now?

Reply:
The command looks like this now: 
```     Command:
      /bin/bash
      -c
      -x
      ceph-osd --foreground --id 0 --fsid 22213e86-aa55-45f1-88aa-9ea7dc4435fb --setuser ceph --setgroup ceph --crush-location="root=default host=minikube" --default-log-to-stderr=true --default-err-to-stderr=true --default-mon-cluster-log-to-stderr=true --default-log-stderr-prefix=debug  --default-log-to-file=false --default-mon-cluster-log-to-file=false --ms-learn-addr-from-peer=false
      rc=$?
      if [ $rc -eq 0 ]; then
        touch /tmp/osd-sleep
        echo "OSD daemon exited with code 0, possibly due to OSD flapping. The OSD pod will sleep for 6 hrs. Restart the pod manually once the flapping issue is fixed"
        sleep 6h
        exit $rc```


Without a quotes `crush-location=root=default host=minikube`, `host=minikube` will be treated as a separate argument and `ceph-osd` command will fail to run as `host=minikube` is not a vaild arg.  So need to use quotes 
`crush-location="root=default host=minikube"`.   

Didn't find any alternatives to this.

Reply:
ah makes sense, thanks for the details